OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -102,17 +158,66 @@ void FindCarryableForTransport(Actor self)
 			}
 		}
 
+		void IResolveOrder.ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString == "DeliverUnit")
+			{
+				var cell = self.World.Map.Clamp(self.World.Map.CellContaining(order.Target.CenterPosition));
+				if (!AircraftInfo.MoveIntoShroud && !self.Owner.Shroud.IsExplored(cell))
+					return;
+
+				if (underAutoCommand)
+				{
+					underAutoCommand = false;
+					self.CancelActivity();
```

Comment:
fixed

Reply:
I'm not sure what you've created, but we already have systems to handle such situations and in the above comment I have have described exactly to use them.

I'm really unsure if this hack solution is even acceptable upstream

Reply:
I don't see its hack.

If you gives order manually which is not 'Stop' then the `busy` flag will be true and `UnderAutoCommand` will be false which will avoid other auto passenger from selecting this transport and change its activity. It will also stop `FerryUnit`.

I don't find any `IsCancelled` check btw in our project. I guess you mean `IsCancelling` but it cannot satisfy the need that prevent auto passenger from selecting this transport.

Reply:
yes, I meant `IsCanceling`. The proper way to interrupt an activity is to exit the `Tick` loop with `true` when `IsCanceling` is true.

Reply:
IsCanceling is only unused in cases where we explicitly don't want an activity to be interrupted, or where it's simply impossible to interrupt like it is for D2K carryalls on bleed. The overriding on Cancel is just an additional orthogonal feature

Reply:
Ok, then what kind of behaiviors are expected on queued order?  `Autocarryall` 's auto action only triggered when the actor is idle, so it will never be the queued activity when you are queueing orders. There are only 2 usecase on queued orders of current `Autocarryall`:

1.  if you queue order before `FerryUnit` pick up the actor, then the carrier will pick up unit then run the queued orders
2.  if you queue order after `FerryUnit` pick up the actor, then the carrier will deliver the unit then run the queued orders

I don't see problems on those two usecases.

Reply:
>  where it's simply impossible to interrupt like it is for D2K carryalls on bleed

`Autocarryall` 's auto action only triggered when the actor is idle, and we have condition here to turn on/off auto action of carryall in this PR.